src/exercises/assignBookmarksToExercises.js

```diff
@@ -165,6 +175,64 @@ function assignBookmarksWithLearningCycle(bookmarks, exerciseTypesList) {
   return exerciseSequence;
 }
 
+function assignBookmarksToLevels(bookmarks, exerciseTypesList) {
+  console.log("bookmarks", bookmarks);
+
+  let exerciseSequence = [];
+
+  const bookmarksByLevel = groupByLevel(bookmarks);
+  const exercisesByLevel = groupByLevel(exerciseTypesList);
+
+  for (let level = 1; level <= 4; level++) {
+    const currentBookmarks = bookmarksByLevel[level] || [];
+    const currentExercises = exercisesByLevel[level] || [];
+
+    let exerciseIndex = 0;
+
+    while (currentBookmarks.length > 0 && currentExercises.length > 0) {
+      const exercise = currentExercises[exerciseIndex];
+      const requiredBookmarks = exercise.requiredBookmarks;
+
+      if (currentBookmarks.length >= requiredBookmarks) {
+        const potentialBookmarks = currentBookmarks.slice(0, requiredBookmarks);
```

Comment:
Don't you run the risk of for example doing a Match with words in different levels? 

Generally, I am thinking shouldn't we try to unify the logic of this with assignBookmarksWithLearningCycle, so they run the same code but just split exercises / bookmarks in a different way? A lot of the code seems to be overlapping.